src/test/java/net/sf/jabref/logic/bibtexkeypattern/MakeLabelWithDatabaseTest.java

```diff
@@ -321,4 +321,57 @@ public void generateKeyAuthIniMany() {
         BibtexKeyPatternUtil.makeLabel(metadata, database, entry, preferences);
         assertEquals(Optional.of("DoeSmiWon"), entry.getCiteKeyOptional());
     }
+
+    public void generateKeyTitleTitleCase() {
+        DatabaseBibtexKeyPattern bibtexKeyPattern = new DatabaseBibtexKeyPattern(pattern);
+        bibtexKeyPattern.setDefaultValue("[title:title_case]");
+        metadata.setBibtexKeyPattern(bibtexKeyPattern);
+        BibtexKeyPatternUtil.makeLabel(metadata, database, entry, preferences);
+        assertEquals(Optional.of("AnAwesomePaperonJabref"), entry.getCiteKeyOptional());
+    }
+
+    @Test
+    public void generateKeyTitleCapitalize() {
+        DatabaseBibtexKeyPattern bibtexKeyPattern = new DatabaseBibtexKeyPattern(pattern);
```

Comment:
The lines `DatabaseBibtexKeyPattern bibtexKeyPattern = ...` and `metadata.setBibtexKeyPattern(bibtexKeyPattern);` can be extracted to the setup method.